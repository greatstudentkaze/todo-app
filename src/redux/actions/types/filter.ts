export const SET_SEARCH_QUERY = 'SET_SEARCH_QUERY';
