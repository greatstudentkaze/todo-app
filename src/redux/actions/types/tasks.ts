export const SET_TASKS = 'SET_TASKS';
export const ADD_TASK = 'ADD_TASK';
export const REMOVE_TASK = 'REMOVE_TASK';
export const CHANGE_TASK_TITLE = 'CHANGE_TASK_TITLE';
export const TOGGLE_TASK = 'TOGGLE_TASK';
