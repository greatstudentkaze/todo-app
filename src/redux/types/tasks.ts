export const SET_TaskS = 'SET_TaskS';
export const ADD_Task = 'ADD_Task';
export const REMOVE_Task = 'REMOVE_Task';
export const CHANGE_Task_TITLE = 'CHANGE_Task_TITLE';
export const TOGGLE_Task = 'TOGGLE_Task';
