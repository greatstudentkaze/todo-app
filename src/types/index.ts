export type TaskItem = {
  id: string,
  title: string,
  completed: boolean,
};
