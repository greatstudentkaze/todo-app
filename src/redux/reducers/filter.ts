import { SET_SEARCH_QUERY } from '../types/filter';
import { FilterActionType } from '../actions/filter';

const initialState = {
  searchQuery: '',
};

export const filter = (state = initialState, action: FilterActionType) => {
  switch (action.type) {
    case SET_SEARCH_QUERY:
      return {
        ...state,
        searchQuery: action.payload,
      }
    default:
      return state;
  }
};
