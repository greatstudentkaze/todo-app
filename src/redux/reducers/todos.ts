import { ToDoItem } from '../../types';
import { ADD_TODO, CHANGE_TODO_TITLE, REMOVE_TODO, SET_TODOS, TOGGLE_TODO } from '../types/todos';
import { ToDosActionType } from '../actions/todos';

const initialState: ToDoItem[] = [
  {
    id: '1',
    title: 'Ознакомиться с TypeScript',
    completed: false,
  },
  {
    id: '2',
    title: 'Сделать компонент списка',
    completed: true,
  },
  {
    id: '3',
    title: 'Сделать компонент todo',
    completed: true,
  },
];

export const todos = (state = initialState, action: ToDosActionType) => {
  switch (action.type) {
    case SET_TODOS:
      return [...action.payload];
    case ADD_TODO:
      return [...state, action.payload];
    case REMOVE_TODO:
      return state.filter(item => item.id !== action.payload);
    case CHANGE_TODO_TITLE:
      return state.map(todo => todo.id === action.payload.id ? {...todo, title: action.payload.title} : todo);
    case TOGGLE_TODO:
      return state.map(todo => todo.id === action.payload ? {...todo, completed: !todo.completed} : todo);
    default:
      return state;
  }
};
