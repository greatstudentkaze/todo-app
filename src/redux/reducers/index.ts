import { combineReducers } from 'redux';

import { filter } from './filter';
import { tasks } from './tasks';

const rootReducer = combineReducers({
  filter,
  tasks,
});

export default rootReducer;
