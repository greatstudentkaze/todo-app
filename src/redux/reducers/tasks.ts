import { ADD_TASK, CHANGE_TASK_TITLE, REMOVE_TASK, SET_TASKS, TOGGLE_TASK } from '../actions/types/tasks';
import { TasksActionType } from '../actions/tasks';
import { Tasks } from '../../types';

const initialState: Tasks = [
  {
    id: '1',
    title: 'Ознакомиться с TypeScript',
    completed: false,
  },
  {
    id: '2',
    title: 'Сделать компонент списка',
    completed: true,
  },
  {
    id: '3',
    title: 'Сделать компонент task',
    completed: true,
  },
];

export const tasks = (state = initialState, action: TasksActionType):Tasks => {
  switch (action.type) {
    case SET_TASKS:
      return [...action.payload];
    case ADD_TASK:
      return [...state, action.payload];
    case REMOVE_TASK:
      return state.filter(item => item.id !== action.payload);
    case CHANGE_TASK_TITLE:
      return state.map(task => task.id === action.payload.id ? {...task, title: action.payload.title} : task);
    case TOGGLE_TASK:
      return state.map(task => task.id === action.payload ? {...task, completed: !task.completed} : task);
    default:
      return state;
  }
};
