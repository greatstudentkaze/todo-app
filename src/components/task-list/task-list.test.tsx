import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import TaskList from './index';
import { Tasks } from '../../types';
import { RootState } from '../../redux/reducers';

const tasks: Tasks = [
  {
    id: '1',
    title: 'Ознакомиться с TypeScript',
    completed: false,
  },
  {
    id: '2',
    title: 'Сделать компонент списка',
    completed: true,
  },
  {
    id: '3',
    title: 'Сделать компонент task',
    completed: true,
  },
];

const filter = {
  searchQuery: '',
};

const initialState: RootState = {
  tasks,
  filter,
};

const mockStore = configureStore([]);

jest.mock('../task-item', () => 'TaskItem');

describe('TaskList component', () => {
  let store;
  let component: renderer.ReactTestRenderer;

  beforeEach(() => {
    store = mockStore(initialState);

    component = renderer.create(
      <Provider store={store}>
        <TaskList />
      </Provider>
    );
  });

  it('renders with given state from mocked Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
