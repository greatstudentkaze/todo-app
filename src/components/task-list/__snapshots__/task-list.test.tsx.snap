// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component renders with given state from mocked Redux store 1`] = `
<ul
  className="task-list"
>
  <TaskItem
    data={
      Object {
        "completed": false,
        "id": "1",
        "title": "Ознакомиться с TypeScript",
      }
    }
  />
  <TaskItem
    data={
      Object {
        "completed": true,
        "id": "2",
        "title": "Сделать компонент списка",
      }
    }
  />
  <TaskItem
    data={
      Object {
        "completed": true,
        "id": "3",
        "title": "Сделать компонент task",
      }
    }
  />
</ul>
`;
